### PR TITLE
fix(universalLogin): create a local copy of flag before signout clears it

### PR DESCRIPTION
**Pull request by ChitlangeSahas**
part of https://github.com/influxdata/idpe/issues/15477

When we tried testing the **logout** portion of the universal login feature, i.e. logout should use `quartz` endpoint and not idpe's, we noticed that the code under the flag `universalLogin` inside `Logout.tsx` was never hit.

After some digging this was the issue in summary:

User clicks logout -> flags cleared -> code looks for the flag value and sees it's false (because it was cleared), so it never runs.

the solution is to save a local copy of the flag on component mount, and then use that instead, so we don't have to worry about flags being cleared.

### Checklist

Authors and Reviewer(s), please verify the following:

- [x] A PR description, regardless of the triviality of this change, that communicates the value of this PR
- [x] [Well-formatted conventional commit messages](https://www.conventionalcommits.org/en/v1.0.0/) that provide context into the change
- [x] Documentation updated or issue created (provide link to issue/PR)
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [ ] Feature flagged, if applicable

### CODE REVIEW

File: src/Logout.tsx [hunk 1]
```diff
@@ -21,10 +21,17 @@ const Logout: FC = () => {
   const history = useHistory()
   const dispatch = useDispatch()
 
+  let universalLoginFlagValue = false
+  useEffect(() => {
+    universalLoginFlagValue = isFlagEnabled('universalLogin')
+  }, [])
```
Comment (ChitlangeSahas) [inline src/Logout.tsx:27]: only run once on mount. mimic `componentDidMount`

File: src/Logout.tsx [hunk 2]
```diff
@@ -21,10 +21,17 @@ const Logout: FC = () => {
...
+
+
   useEffect(() => {
     const handleSignOut = async () => {
       if (CLOUD) {
-        if (isFlagEnabled('universalLogin')) {
+        console.warn('universalLoginFlagValue: ', universalLoginFlagValue)
+        if (universalLoginFlagValue) {
```
Comment (ChitlangeSahas) [inline src/Logout.tsx:34]: use the saved flag value

File: src/Logout.tsx [hunk 3]
```diff
@@ -21,10 +21,17 @@ const Logout: FC = () => {
...
+  useEffect(() => {
+    universalLoginFlagValue = isFlagEnabled('universalLogin')
+  }, [])
+
+
   useEffect(() => {
     const handleSignOut = async () => {
       if (CLOUD) {
-        if (isFlagEnabled('universalLogin')) {
+        console.warn('universalLoginFlagValue: ', universalLoginFlagValue)
```
Comment (ChitlangeSahas) [inline src/Logout.tsx:33]: just for sanity test. i will clean the warns once we decide to remove the flag.

Comment (ChitlangeSahas) [reply src/Logout.tsx:33]: This is hard to test locally, because of weird interactions with dex.

**Comment by wdoconnell** [comment]
FYI, I'm reviewing this now.

**Comment by hoorayimhelping** [comment]
>User clicks logout -> flags cleared -> code looks for the flag value and sees it's false (because it was cleared), so it never runs.

It looks to me like the flow is: user clicks logout -> `handleSignout` function is created -> `dispatch(reset())` is called -> `handleSignout` is called -> during this call, flag is checked but it has just been reset

Would it be simpler to  call `dispatch(reset())` after checking the state of the flag?

**Comment by ChitlangeSahas** [comment]
> > User clicks logout -> flags cleared -> code looks for the flag value and sees it's false (because it was cleared), so it never runs.
> 
> It looks to me like the flow is: user clicks logout -> `handleSignout` function is created -> `dispatch(reset())` is called -> `handleSignout` is called -> during this call, flag is checked but it has just been reset
> 
> Would it be simpler to call `dispatch(reset())` after checking the state of the flag?

@hoorayimhelping : Me and bill were thinking the same, on friday, the only issue is the handleSignout will redirect to quartz codebase, and will no longer continue to execute the ui code. (logout happens upon window.replace to a quartz url)

this is why we thought it may not work.

**Comment by hoorayimhelping** [comment]
>the only issue is the handleSignout will redirect to quartz codebase, and will no longer continue to execute the ui code. (logout happens upon window.replace to a quartz url)

If it's no longer executing UI code, why does it matter if `reset()` is called?

**Comment by wdoconnell** [comment]
> > the only issue is the handleSignout will redirect to quartz codebase, and will no longer continue to execute the ui code. (logout happens upon window.replace to a quartz url)
> 
> If it's no longer executing UI code, why does it matter if `reset()` is called?

The `reset()` is needed for a fix that was applied here:
https://github.com/influxdata/ui/commit/ae9f199f6d9783bede66f573619f493b474d053f
https://github.com/influxdata/ui/pull/176

I don't believe the OSS login flow has changed since then in a way that would permit removing the reset.

**Comment by wdoconnell** [comment]
FYI @hoorayimhelping - Sahas is going to move the conditional into the same useEffect hook.

**Comment by hoorayimhelping** [comment]
@wdoconnell 

>The `reset()` is needed for a fix that was applied here:
>https://github.com/influxdata/ui/commit/ae9f199f6d9783bede66f573619f493b474d053f
>https://github.com/influxdata/ui/pull/176

Are you sure? That change added

```ts
dispatch({type: 'USER_LOGGED_OUT'})
```

```ts
dispatch(reset())
```

was already there. `reset()` clears the feature flag state and is what is applicable to this PR.

**Comment by wdoconnell** [comment]
I'm not familiar with how to access PRs from this time period - looks like another repo? https://github.com/influxdata/ui/commit/43ba458e8cf533725108470ce768b6c300803679

**Comment by hoorayimhelping** [comment]
https://github.com/influxdata/influxdb/pull/17926

**Comment by wdoconnell** [comment]
(For context - that's the commit that adds the feature flag `reset`. But not linked to a PR we can use to ascertain the reason for the change.)

**Comment by wdoconnell** [comment]
Let me ask Ari, since I think he reviewed that PR. It's a little hard to ascertain what the background behind adding the `reset()` is. This is a while back but maybe he'll remember. 😄

**Review by wdoconnell — APPROVED** [review]

**Pull request merged**